File: C++/Problems/Problems/problem_1750.cpp
#include "problem_1750.h"

bool problem_1750::test() {
    std::string s = "ca";

    int expected = 2;

    auto result = minimumLength(s);

    std::cout << result << std::endl;

    return expected == result;
}

int problem_1750::minimumLength(std::string s) {
    int n = s.size();
    int l = 0;
    int r = n - 1;
    while (l < r && l < n && r >= 0 && s[l] == s[r])
    {
        char c = s[l];
        while (l < n && s[l] == c)
            l++;
        while (r >= 0 && s[r] == c)
            r--;
    }

    if (l < r)
        return r - l + 1;
    return l == r ? 1 : 0;
}

File: Python/Problems/Problems/problem1750.py
import heapq
import math
import sys

from utils import print1DArray, print2DArray
from typing import List, Set
from collections import Counter

from .problem import Problem


class Problem1750(Problem):
    def test(self) -> bool:
        s = "ca"

        expected = 2

        result = self.minimumLength(s)

        print(result)

        return expected == result

    def minimumLength(self, s: str) -> int:
        n = len(s)
        l = 0
        r = n - 1
        while l < r and l < n and r >= 0 and s[l] == s[r]:
            c = s[l]
            while l < n and s[l] == c:
                l += 1
            while r >= 0 and s[r] == c:
                r -= 1

        if l < r:
            return r - l + 1
        return 1 if l == r else 0